translate/Main.hs

```diff
@@ -114,6 +114,36 @@ doTranslate signs circuit = do
             when (undef  /= []) $ putStr $ "The following signals have undefined initial states: \n"
                                     ++ unlines (map show undef) ++ "\n"
 
+handleArcs :: [([Transition DynSignal], Transition DynSignal)] -> [String]
+handleArcs arcLists
+    | arcLists == [] = []
+    | otherwise = do
+    let effect = snd (head arcLists)
+    -- Filter for just one of the effects
+    let causesForEffect = filter (\o -> snd o == effect) arcLists
+    -- Get just the list of lists of causes
+    let causesLists = map (\l -> fst l) causesForEffect
+    -- Remap these
+    let mapped = sequence causesLists
+    -- Length of this list of lists is the number of extra transitions to be added
+    let n = length mapped
+    -- Create arc pairs for these arcs
+    let arcs = arcarcs mapped effect n
+    -- Get remaining arcs for other effects
+    let remainder = arcLists \\ causesForEffect
+    -- return [consistency loop additions] ++ [arc strings] ++ [the same result for other effects]
+    addSymbTransition effect n ++ concatMap transition arcs ++ handleArcs remainder
+
+addSymbTransition :: Transition DynSignal -> Int -> [String]
+addSymbTransition effect n
+        | "+" `isSuffixOf` (show effect) = map (\x -> (printf "%s0 %s/%s\n" (init (show effect)) (show effect) (show x)) ++ (printf "%s/%s %s1" (show effect) (show x) (init (show effect)))) [1..n - 1]
```

Comment:
What are you trying to achieve with `"+" `isSuffixOf` (show effect)`? Is this simply `newValue effect == True` or, even shorter, `newValue effect`?